Source/Infrastructure/Read/IExtendedReadModelRepositoryFor.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Dolittle.ReadModels;
+using MongoDB.Driver;
+
+namespace Infrastructure.Read
+{
+    public interface IExtendedReadModelRepositoryFor<T> : IReadModelRepositoryFor<T>
+        where T : IReadModel
```

Comment:
A big interface, but it provides most of the same methods that IMongoCollection provides.